### DISCUSSION TITLE
strongSwan on iOS

### DISCUSSION THREAD

**Discussion by kojimr** [strongswan/strongswan#877] [Q&A] [answered]
Is it possible to run strongSwan on iOS?
I don't think there is a description of how to install it, although the website states the following.

[https://www.strongswan.org/](https://www.strongswan.org/)
> runs on Linux 2.6, 3.x and 4.x kernels, Android, FreeBSD, OS X, iOS and Windows

**Comment by tobiasbrunner** [comment, marked as answer]
The strongSwan libraries can be used to write IKEv2/IPsec-based VPN applications that run on iOS. We currently don't provide such an app ourselves, though.

**Reply by kojimr** [reply]
Thanks for the reply. I understand that the library can be used on iOS.
Is there any documentation on how to use the strongSwan library on iOS? Please let me know if there is.

**Reply by tobiasbrunner** [reply]
No, currently not. It's similar to our Android app, though, as it also requires a component like libandroidbridge that ties the libraries to the app and e.g. creates configs, controls SAs and manages TUN devices via [Packet Tunnel Provider](https://developer.apple.com/documentation/networkextension/packet_tunnel_provider). The [README for our old macOS app](https://github.com/strongswan/strongswan/blob/master/src/frontends/osx/README.md) might also be relevant as building the libraries and using them in Xcode is similar for iOS.

**Reply by kojimr** [reply]
Thank you for the information. I will give it a try.

**Reply by dalekzhangdong** [reply]
you can write a ios app with ios [build in vpn protocols](https://developer.apple.com/documentation/networkextension/personal_vpn). I use ios Personal VPN(ikev2 psk ) to write a ios app, and test with strongswan, it's fine.

**Reply by Young1006** [reply]
Do you know how to use it on iOS now? I'm confused, too.

**Reply by MewanCdot** [reply]
> Thank you for the information. I will give it a try.

Hi @kojimr , I am trying to build the strongSwan app on iOS but with oqs support, have you progressed on building the strongswan libraries for iOS, or an iOS equivalent for libandroidbridge?
